go/api/stream_commands_test.go

```diff
@@ -0,0 +1,791 @@
+
+	client.XGroupCreateWithOptions(key, group, "0", options.NewXGroupCreateOptions().SetMakeStream())
+	client.XGroupCreateConsumer(key, group, consumer)
+	client.XAddWithOptions(
+		key,
+		[][]string{{"entry1_field1", "entry1_value1"}, {"entry1_field2", "entry1_value2"}},
+		options.NewXAddOptions().SetId(streamId),
+	)
+
+	response, err := client.XReadGroup(group, consumer, map[string]string{key: "0"})
+	if err != nil {
+		fmt.Println("Glide example failed with an error: ", err)
+	}
+	fmt.Println(response)
+
+	// Output: map[12345:map[]]
+}
+
+func ExampleGlideClient_XReadGroupWithOptions() {
+	var client *GlideClient = getExampleGlideClient() // example helper function
+	key := "12345"
+	streamId := "12345-1"
+	group := uuid.NewString()
+	consumer := uuid.NewString()
+
+	client.XGroupCreateWithOptions(key, group, "0", options.NewXGroupCreateOptions().SetMakeStream())
+	client.XGroupCreateConsumer(key, group, consumer)
+	client.XAddWithOptions(
+		key,
+		[][]string{{"entry1_field1", "entry1_value1"}, {"entry1_field2", "entry1_value2"}},
+		options.NewXAddOptions().SetId(streamId),
+	)
+
+	options := options.NewXReadGroupOptions().SetNoAck()
+	response, err := client.XReadGroupWithOptions(group, consumer, map[string]string{key: ">"}, options)
+	if err != nil {
+		fmt.Println("Glide example failed with an error: ", err)
+	}
+	fmt.Println(response)
+
+	// Output: map[12345:map[12345-1:[[entry1_field1 entry1_value1] [entry1_field2 entry1_value2]]]]
+}
+
+func ExampleGlideClient_XRead() {
+	var client *GlideClient = getExampleGlideClient() // example helper function
+	key := "12345"
+	streamId := "12345-1"
+
+	client.XAddWithOptions(
+		key,
+		[][]string{{"field1", "value1"}, {"field2", "value2"}},
+		options.NewXAddOptions().SetId(streamId),
+	)
+
+	response, err := client.XRead(map[string]string{key: "0-0"})
+	if err != nil {
+		fmt.Println("Glide example failed with an error: ", err)
+	}
+	fmt.Println(response)
+
+	// Output: map[12345:map[12345-1:[[field1 value1] [field2 value2]]]]
+}
+
+func ExampleGlideClient_XReadWithOptions() {
+	var client *GlideClient = getExampleGlideClient() // example helper function
+	key := "12345"
+	streambase := 1
+
+	genStreamId := func(key string, base int, offset int) string { return fmt.Sprintf("%s-%d", key, base+offset) } // helper function to generate stream ids
+
+	client.XAddWithOptions(
+		key,
+		[][]string{{"field1", "value1"}, {"field2", "value2"}},
+		options.NewXAddOptions().SetId(genStreamId(key, streambase, 0)),
+	)
+	client.XAddWithOptions(
+		key,
+		[][]string{{"field3", "value3"}, {"field4", "value4"}},
+		options.NewXAddOptions().SetId(genStreamId(key, streambase, 1)),
+	)
+
+	response, err := client.XReadWithOptions(map[string]string{key: genStreamId(key, streambase, 0)}, options.NewXReadOptions().SetCount(1))
+	if err != nil {
+		fmt.Println("Glide example failed with an error: ", err)
+	}
+	fmt.Println(response)
+
+	// Output: map[12345:map[12345-1:[[field1 value1] [field2 value2]]]]
```

Comment:
Is it possible to post sample output in a more readable format?

Reply:
Let me look at the options with JSON to format the output.

Reply:
Will look at this after we get it into 1.3 branch.  We are undergoing too much churn on this branch since the PR is so large that every other checkin keeps blocking this PR.